src/parquet/column/levels.h

```diff
@@ -123,8 +132,8 @@ class LevelDecoder {
   // Decodes a batch of levels into an array and returns the number of levels decoded
   size_t Decode(size_t batch_size, int16_t* levels) {
     size_t num_decoded = 0;
-    if (!rle_decoder_ && !bit_packed_decoder_) {
-      throw ParquetException("Level decoders are not initialized.");
+    if (num_values_remaining_ == 0) {
+      throw ParquetException("Level decoders are not initialized with data.");
     }
```

Comment:
If the level decoders are not initialized either due to empty data pages or "required" data, the semantics of `Decode` should simply return 0 elements. If on the other hand, the decoders are initialized, then an exception must be thrown for `Decode` on zero remaining elements. If we want to simplify, we could return zero on both these cases.  Does this sound reasonable ?